### PR TITLE
Invalid successful alert is displayed when a bad note is added

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | In the BO > Orders > Order details page > when we add an order note with invalid, a Update successful alert is displayed by the note is empty.
| Type?             | bug fix 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/24534
| How to test?      | See issue
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24582)

### CODE REVIEW

**Comment by okom3pom** [comment]
Possible better wording ? if you can add the Waiting for wording label ?

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 1]
```diff
@@ -956,12 +956,14 @@ public function removeCartRuleAction(int $orderId, int $orderCartRuleId): Redire
      */
     public function updateInvoiceNoteAction(int $orderId, int $orderInvoiceId, Request $request): RedirectResponse
     {
-        $this->getCommandBus()->handle(new UpdateInvoiceNoteCommand(
+        if ($this->getCommandBus()->handle(new UpdateInvoiceNoteCommand(
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
        try {
            $this->getCommandBus()->handle(new UpdateInvoiceNoteCommand(
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 2]
```diff
@@ -956,12 +956,14 @@ public function removeCartRuleAction(int $orderId, int $orderCartRuleId): Redire
...
             $orderInvoiceId,
             $request->request->get('invoice_note')
-        ));
-
-        $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+        ))) {
+            $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+        } else {
+            $this->addFlash('error', $this->trans('Invalid characters:', 'Admin.Notifications.Info'));
+        }
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
            )));
            $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
        } catch(InvoiceException $e) {
            $this->addFlash('error', $this->trans('Invalid characters:', 'Admin.Notifications.Info'));
        }
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Better use try catch as the Handler for `src/Adapter/Order/CommandHandler/UpdateInvoiceNoteHandler.php` doesn't return nothing.

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 3]
```diff
@@ -956,12 +956,15 @@ public function removeCartRuleAction(int $orderId, int $orderCartRuleId): Redire
      */
     public function updateInvoiceNoteAction(int $orderId, int $orderInvoiceId, Request $request): RedirectResponse
     {
-        $this->getCommandBus()->handle(new UpdateInvoiceNoteCommand(
+        try {
+            $this->getCommandBus()->handle(new UpdateInvoiceNoteCommand(
             $orderInvoiceId,
             $request->request->get('invoice_note')
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
                $orderInvoiceId,
                $request->request->get('invoice_note')
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 4]
```diff
@@ -956,12 +956,15 @@ public function removeCartRuleAction(int $orderId, int $orderCartRuleId): Redire
...
-        ));
-
-        $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+            )));
+            $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+        } catch(InvoiceException $e) {
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:966]: Could you check that `use PrestaShop\PrestaShop\Core\Domain\Order\Invoice\Exception\InvoiceException;` is put on the top of the file ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Comment by ghost** [comment]
Hey @Progi1984 i test with `try` but i don't have an error when i use < > in the note with the `if` i have the error.

I add use PrestaShop\PrestaShop\Core\Domain\Order\Invoice\Exception\InvoiceException; at the top.

Thank you for learning

**Comment by ghost** [comment]
I made a mistake because the regression is for Order Note and not Invoice Note ...  so the branch 178 is not correct i close this PR and i made a new one in the good branch

**Pull request closed without merging**